backend/common/src/util/null.rs

```diff
@@ -0,0 +1,136 @@
+use serde::de::{Deserialize, Deserializer, IgnoredAny};
+use serde::ser::{Serialize, Serializer};
+
+/// Alternative to `serde::de::IgnoreAny` that implements `Serialize`.
+/// Will serialize to `null` in JSON, or empty data in bincode.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct NullAny;
```

Comment:
I believe this can go away with separate json and non-json `Payload`.

Reply:
I think so too; there are a couple of things I think I'll be able to remove when I tidy up the common things :)